### PR TITLE
ODR-808 - RackHD SKU packs require bmcMac keyword to provide MAC address of BMC

### PR DESCRIPTION
**Pull request by larry-dean**
- dell-c6320
- dell-r630
- quanta-t41
- quanta-d51-1u
- quanta-d51-2u

### CODE REVIEW

**Comment by larry-dean** [comment]
@RackHD/corecommitters @jimturnquist

**Comment by heckj** [comment]
@larry-dean not sure you meant to change the permissions on all these files from 0644 to 0755... is that intentional?

**Comment by larry-dean** [comment]
I'll fix that

**Comment by larry-dean** [comment]
Changes pushed up.

File: quanta-t41/config.json [hunk 1]
```diff
@@ -31,11 +31,11 @@
             "md5sum": "f1da3b9bce73155716de6c3959b43219",
             "command": "bios/SCELNX_64",
             "args": "/i /s ${FLASH_FILE}"
-        }
+        },
+         "bmcMac": "rmm.data.MAC Address"
```
Comment (jimturnquist) [inline quanta-t41/config.json]: I believe this should be:

'bmc_mac': '#bmc.data.MAC Address'

File: dell-c6320/config.json [hunk 2]
```diff
@@ -10,6 +10,12 @@
             "contains": "04FNTC"
         }
     ],
+    "skuConfig" : {
+        "redfish" : {
+            "computeType": "Blade"
+        },
+        "bmcMac": "rmm.data.MAC Address"
```
Comment (jimturnquist) [inline dell-c6320/config.json]: Should be:

'bmcMac': 'bmc.data.MAC Address'

File: dell-r630/config.json [hunk 3]
```diff
@@ -10,6 +10,12 @@
             "contains": "0CNCJW"
         }
     ],
+    "skuConfig" : {
+        "redfish" : {
+            "computeType": "Rackmount"
+        },
+        "bmcMac": "rmm.data.MAC Address"
+    },
```
Comment (jimturnquist) [inline dell-r630/config.json:18]: Should be:

'bmcMac': 'bmc.data.MAC Address'

File: dell-r730/config.json [hunk 4]
```diff
@@ -10,6 +10,12 @@
             "contains": "072T6D"
         }
     ],
+    "skuConfig" : {
+        "redfish" : {
+            "computeType": "Rackmount"
+        },
+        "bmcMac": "rmm.data.MAC Address"
```
Comment (jimturnquist) [inline dell-r730/config.json]: Should be:

'bmcMac': 'bmc.data.MAC Address'

File: dell-r730xd/config.json [hunk 5]
```diff
@@ -10,6 +10,12 @@
             "contains": "04N3DF"
         }
     ],
+    "skuConfig" : {
+        "redfish" : {
+            "computeType": "Rackmount"
+        },
+        "bmcMac": "rmm.data.MAC Address"
```
Comment (jimturnquist) [inline dell-r730xd/config.json]: Not sure where bmcMac is found on this Dell.  If it follows prior Dell art, then:

'bmcMac': 'bmc.data.MAC Address'

File: quanta-d51-1u/config.json [hunk 6]
```diff
@@ -25,10 +25,7 @@
...
             "command": "bios/afulnx64/afulnx_64",
             "args": "${FLASH_FILE} /P /B /K /N /L"
-        }
-    },
-    "workflowRoot": "workflows",
-    "taskRoot": "tasks",
-    "httpTemplateRoot": "templates",
-    "httpStaticRoot": "static"
+        },
+        "bmcMac": "rmm.data.MAC Address"
```
Comment (jimturnquist) [inline quanta-d51-1u/config.json]: Should be:

'bmcMac': 'bmc.data.MAC Address'

File: quanta-d51-2u/config.json [hunk 7]
```diff
@@ -25,10 +25,10 @@
             "md5sum": "f4eb7bfea73accc32860425f14f4aa6d",
             "command": "bios/afulnx64/afulnx_64",
             "args": "${FLASH_FILE} /P /B /K /N /L"
-        }
+        },
+        "bmcMac": "rmm.data.MAC Address"
```
Comment (jimturnquist) [inline quanta-d51-2u/config.json]: Should be:

'bmcMac': 'bmc.data.MAC Address'

**Comment by jimturnquist** [comment]
The bmcMac assignments should be taken from product_mapper.conf

**Comment by larry-dean** [comment]
Changes pushed up.

**Comment by zyoung51** [comment]
The removal of static in the quanta sku packs results in the creation of skupacks where the firmware upgrade workflows will not work.  The Readme in those directories specifies how the static directory should be populated with vendor supplied files.

**Comment by derrickostertag** [comment]
@derrickostertag

**Comment by larry-dean** [comment]
Changes pushed up.

**Comment by yyscamper** [comment]
@larry-dean: Base on your code change, I think the commit message and PR title need some change.

**Comment by zyoung51** [comment]
@yyscamper @RackHD/corecommitters For the Quanta (and Platform-R) the README details the steps to populate the static folder with vendor supplied artifacts that we cannot distribute, but this is the 2nd PR that has attempted to fix the required, and intentionally missing, 'staticRoot' folder causing an issue.  Any suggestions on how to improve the logic here?

**Comment by jimturnquist** [comment]
So, the change in its current form is adding bmcMac to all appropriate skupacks and adding in the redfish compute type where necessary.

+1

**Comment by zyoung51** [comment]
+1

**Comment by yyscamper** [comment]
@zyoung51 : Leave the `static` folder as empty to ensure the integrity of skupack is not straightforward and less obvious, user tends to neglect it and confuse when found inserting skupack failure.

The two quick solutions that come to my mind are as following:

(1) improve the build script, define particular build script for quanta d51, so if any file is missing, or the FW image doesn't match with the md5,  you can report error message in console and reject building skupack.

(2) Create manifest file within skupack, when build skupack, validate all items that listed in manifest file, report error in build level if validation failure. Inside RackHD, it can also do the same validation, so RackHD will reject the skupack insertion if manifest validation fails.

**Comment by yyscamper** [comment]
For PR itself: :+1:

**Comment by yyscamper** [comment]
@zyoung51 : Let's move out your question to a google group discussion or an enhancement issue.

**Pull request merged**